---
title: 'Project 3: Ovarian Cancer'
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```
## Contents
1. Overview Ovarian Cancer
2. Data Cleanup
3. Driver Mutations
4. co-existing Mutations
5. Synthetic Lethality Interactions
6. Linear Regression
7. Summary of Results
8. Conclusion

## 1. Overview Ovarian Cancer

### 1.1 General facts

Ovarian cancer is the one of the leading cancer death causes in women. One of 71 women will be diagnosed with this disease at some point during her lifetime. When treated in early stages, OC has relatively high survival rates. However, symptomes mostly start to develop when the tumour has already progressed to advanced stages, which is why the overall 5-year survival rate for OC lies under 50%.

Ovarian cancer is a very heterogenous disease and mostly represented by epithelial tumours. There are four main subtypes: serous, endometrioid, mucinous and clear cell. Furthermore, tumours can be classified dependent on their state of differentiation. Low grade (LG) Ovarian cancer is well differentiated and therefore not invasive; high grade (HG) tumours are often aggressive and frequently metastasize.

Standard therapy approaches include surgery and chemotherapy with DNA-damaging agents. Personalized immunotherapeutic approaches often target special features of cancer cell genome and proteome such as over-expressed surface receptors.

### 1.2 Common mutations in OC (literature)

Literature on Ovarian cancer states that the most common mutations are found in the DNA repair-related genes BRCA1 and BRCA2. Abnormalities in other DNA and mismatch repair genes have been found too, as well as TP53, which is also known as "guardian of the genome". According to Testa et al., TP53 accounts for most mutations found in high grade serous Ovarian cancer.
It will be our task to verify this information and to gain insight on potential driver mutations in Ovarian cancer.

## 2. Data Cleanup

first we downloaded the data and load data in R
```{r}
allDepMapData <- readRDS("~/GitHub/project-01-group-03/DepMap19Q1_allData.RDS")
```
rename the lists from the main data
```{r}
copynumber = allDepMapData[["copynumber"]]
mutation = allDepMapData[["mutation"]]
kd.ceres = allDepMapData[["kd.ceres"]]
kd.prob = allDepMapData[["kd.prob"]]
annotation = allDepMapData[["annotation"]]
expression = allDepMapData[["expression"]]

```
extract all cell lines that have ovarian cancer as primary disease. 
```{r}
annotation = annotation[which(annotation$Primary.Disease == "Ovarian Cancer"), ] # The annotation matrix now only consists of Ovarian Cancer cell lines
ID = annotation$DepMap_ID # DepMap_ID is the column name for the Ovarian Cancer cell lines
expression = expression[ , which(colnames(expression) %in% ID)] # Extract the expression column from ID and re-define the data frame called expression. Do the same for the columns copynumber, kd.ceres and kd.prob. (Stimmt das?)
copynumber = copynumber [ , which(colnames(copynumber) %in% ID)]
kd.ceres = kd.ceres [ , which(colnames(kd.ceres) %in% ID)]
kd.prob = kd.prob [ , which(colnames(kd.prob) %in% ID)]
mutation = mutation [ ID]
```


remove the object allDepMapData for a clean workflow
```{r}
rm(allDepMapData)
```

Remove unnecessary cols of the annotation matrix and rename the rows with the cell lines
```{r}
rownames(annotation) = annotation$DepMap_ID
annotation = annotation[, -which(colnames(annotation) %in% c("DepMap_ID", "Aliases", "Primary.Disease", "Gender", "Source"))]

```

check the matrices if they contain any NA values and delete the containing rows

```{r}
#summary(copynumber) hier m?ssen wirs hinkriegen dass er den output net reinhaut


NAV = apply(copynumber, 1, function(x) {sum(is.na(x))}) 
copynumber = copynumber[-which(NAV > 0), ]  

#hier muss ich nochmal gucken ob man des machen muss weil iwi passiert da was komisches

#NAV = apply(expression, 1, function(x) {sum(is.na(x))}) 
#expression = expression[-which(NAV > 0), ]  
 
#NAV = apply(kd.ceres, 1, function(x) {sum(is.na(x))}) 
#kd.ceres = kd.ceres[-which(NAV > 0), ]

#NAV = apply(kd.prob, 1, function(x) {sum(is.na(x))}) 
#kd.prob = kd.prob[-which(NAV > 0), ]
```


## 3. Driver Mutations


Fuse mutation lists to one matrix
```{r}
mutation.all = as.data.frame(rbindlist(mutation))
```

Our analyses just need the information in certain columns of our data frame, for example gene name and location (chromosome), cell line, the kind of mutation (missense, frame shift, etc.). Thus, we extract these columns and put them in the data frame "mutation.all".
```{r}
 mutation.all = mutation.all[, which(colnames(mutation.all) %in% c("Hugo_Symbol", "DepMap_ID", "Variant_Classification", "Variant_annotation", "isTCGAhotspot", "Chromosome", "isDeleterious"))]
```
There are different types of mutations. Some of them are silent, which means that the amino acid sequence is not altered and the protein structure is not affected. Other mutations lead to an amino acid exchange, but do not provoke dramatic conformational changes in the encoded protein. Whether or not a mutation has an impact on protein function is noted in the column "isDeleterious". We want to extract all mutations that are TRUE for isDeleterious, since these might have something to do with cancer cell development.
```{r}
mutation.all = mutation.all[which(mutation.all$isDeleterious == "TRUE"), ] # only include rows (=genes) that have deleterious mutations in data frame "mutation.all"
```
Order matrix alphabetically

```{r}
mutation.all <- mutation.all[order(mutation.all$Hugo_Symbol),]

```

Find most frequently mutated genes by summing up all "mutation events" found in all cell lines. To do this, the names of the genes were converted to factors.
```{r}
mutation.all$Hugo_Symbol = factor(mutation.all$Hugo_Symbol)

summary(mutation.all$Hugo_Symbol)

```
Plot all mutations to see most frequently mutated genes
```{r}
plot(mutation.all$Hugo_Symbol)
```


The most common mutation is found in ARID1A (16 times). Surprisingly, BRCA1 and BRCA2 are only mutated twice and once, respectively.

Problem: Many mutations occur several times in one cell line, but in different DNA loci. If we want to identify the most frequent mutations among all cell lines, these duplicates should not be included. We want to extract mutations that occur frequently in different cell lines. Therefore, we use the "duplicated" function from the dplyr package and define a new data frame called new_uniq that only contains one mutation of a gene per cell line for simplicity (since the type of mutation is not relevant here).

```{r}
duplicates <- which(duplicated(mutation.all[c('Hugo_Symbol', 'DepMap_ID')]), ) #find all identical combinations of mutated gene and cell line
duplicates # show all genes that are duplicated
new_uniq <- mutation.all[!duplicated(mutation.all[c('Hugo_Symbol', 'DepMap_ID')]),] #do not include duplicates in data frame new_uniq
summary(new_uniq)

```
As we can see, ARID1A is only the second most common mutation after TP53, which is mutated in 12 of 34 different cell lines. The most common mutations among our cell lines are thought to be our driver mutations. We will be conducting further analyses with these genes.

For a better overview, we solely plot genes that are mutated more than 4 times in total. Further analyses will focus on these designated driver mutations.

```{r}
DriverMutation <- as.data.frame(table(new_uniq$Hugo_Symbol)) # Create a new data frame that contains the mutated genes (but just once per cell line, as definded in new_uniq)
DriverMutation = DriverMutation[which(DriverMutation$Freq > 4), ] # extract all mutations that occur more than 4 times among our cell lines in total
DriverMutation <- DriverMutation[order(DriverMutation$Freq, decreasing = TRUE),] # order the genes in the data frame according to their frequency
barplot(DriverMutation$Freq, names.arg = DriverMutation$Var1, xlab="mutated Gene", las = 1, horiz=FALSE, cex.axis = 1,2, col = "lightblue", main = "mutations present in more than four cell lines", ylab = "frequency") # create a barplot of the most frequent mutations, displaying their occurence


```



## 4. Co-existing Mutations

```{r}
library(dendextend)  # we need this package to create tree diagrams

tree <- expression[which(rownames(expression) %in% DriverMutation$Var1), ]
#neue Matrix belegen, damit expression nicht überschribene wird
# aber komischerweise fehlt ein mutiertes Gen in der Expression BAI... ???

cor.mat = cor(tree, method = "spearman")
cor.dist = as.dist(1 - cor.mat)
cor.hc = hclust(cor.dist, method = "ward.D2")
cor.hc = as.dendrogram(cor.hc)


plot(cor.hc, las = 2, cex.lab = 0.7)


cb1 <- ifelse(annotation == "Adenocarcinoma", "forestgreen", "firebrick")
cb <- cbind(`Adeno` = cb1)
plot(cor.hc, las = 2, cex.lab = 0.7)
colored_bars(colors = cb, dend = cor.hc)

# David fragen warum das nicht funktioniert, beim 3FS Kurs hats geklappt
#http://bioinfo.ipmb.uni-heidelberg.de/crg/datascience3fs/practicals/2_UnsupervisedLearning.html#hierarchical-clustering

```

Here, we want to investigate whether there are certain mutations that often occur together and are therefore possibly linked to each other in their genesis. The goal will be to create a symmetrical matrix whose columns AND rows are our genes; the cells' content will be the counts of how often two corresponding genes turn up in the same cell line together. Next, maximum values can be extracted and tested for significance.




## 5. Synthetic Lethality Interactions
After identifying our most common driver mutations, we used this information to find possible synthetic lethality interaction partners
First of all, we want to calculate the variance of the cell survival (kd.ceres) and estimated cell survival probability (kd.prob) when knocking out specific genes. We hereby assume that genes that display high variances when knocked out are more likely to have specific synthetic lethality interactions since genes that broadly lead to cell death in most cell lines are probably essential for survival on their own. Therefore, we will conduct the search for synthetic lethality interaction partners with genes whose variances are greater than the 75% quantile.

```{r}
topVar = apply(kd.prob, 1, var) # calculate variance over all rows (genes)
summary(topVar)
kd.prob.topVar = kd.prob[topVar > quantile(topVar, probs = 0.75), ] # new data frame with just the genes whose knockout leads to highly variant cell death events -> perhaps our synthetic lethality interaction partners?
dim(kd.prob.topVar) # show me how many genes we are looking at now

mean.survival.prob <- apply(kd.prob.topVar, 1, mean) # calculate the mean cell survival probability of the remaining genes
summary(mean.survival.prob) # show me the mean survival distribution of the genes whose knockout leads to the most variant cell survival

plot(density(as.matrix(mean.survival.prob))) # plot me the distribution of cell survival probability of most variant survivals

#weiß noch nicht, wie sinnvoll das alles ist :D mal sehen


```


## 6. Linear Regression
In this step we wanted to investigate if it is possible to estimate the gene expression of a gene based on the copynumber. First a linear regression model should show if this issue can be applied to the totality of our data set independent of the differences between the single genes. Another regression model will focus on the correlation between the seven driver mutation
Last we want to investigate the relationship between the gene expression and copy number for each driver mutation individually.

### 6.1 Regression model for all genes

#### 6.1.1 Structuring the data

In the first step the data frames copynumber, knock.down and expression were ordered alphabetically, this helps to gain a better overview of the reviewed genes.

```{r}
copynumber <- copynumber[order(rownames(copynumber)),]
expression <- expression[order(rownames(expression)),]
kd.ceres <- kd.ceres[order(rownames(kd.ceres)),]
kd.prob <- kd.prob[order(rownames(kd.prob)),]
```
Only genes, who are present in all dataframes, can be included for the linear regression model. There for we first established a list, which contains all three dataframes. This step allows to extract the shared genes in both dataframes in a more effectiv way.
```{r}
list_all.genes <- list(expression,copynumber, kd.ceres, kd.prob ) # bind all the dataframes in one list
names(list_all.genes) <- c("expression","copynumber", "kd.ceres", "kd.prob" ) # rename the elements of the list
```
Next we will only extraxt genes, that are common in all three dataframes. Therefore all identical rownames/ genes were collated in one object/value
```{r}
common_names = Reduce(intersect, lapply(list_all.genes, row.names)) # all common rownames in the three dataframes
list_all.genes <- lapply(list_all.genes, function(x) {x[row.names(x) %in% common_names,]}) # new list reduced to only the shared genes/ rownames
```
To check if the process was succesful, the dimensions before and after the clean up process were compared.
```{r}
dim(expression)
dim(copynumber)
dim(kd.ceres)
dim(kd.prob)
```
The dimension of the primary dataframes shows 49070, 23299  and 17634 rows, which indicates the present of different genes in the dataframes. After the clean up all data frames contain 16970 genes and the 34 ovarian cancer cell lines
```{r}
dim(list_all.genes$expression)
dim(list_all.genes$copynumber)
dim(list_all.genes$kd.ceres)
dim(list_all.genes$kd.prob)

```

#### 6.1.2 Checking distribution 
##### Gene expression
As a requirement for the regression model the data sets have to show a normal distribution. For an visual assessment  a Q-Q-plot was used for each of the four variables. First all the expression values of the different cell lines are fused to one long column. 
For an easierer application of a Q-Q-plot, the package reshape
```{r}
install.packages("reshape")
library(reshape)
```
After the installation the values of each cell lines are fused to a new matrix exp1, which contains only two columns. The first column shows the original cell line of the listed expression value in the second column.
```{r}
exp1 <- melt.data.frame(list_all.genes$expression, variable_name = "cell line") #fuse all expression values to one long column
```
The new matix will be used to check the distribution of the gene expression. The q-q-plot compairs the quantils of an normal distribution with the distribution of our data set and therefor can be used to check normality of the expression data
```{r}
qqnorm(exp1$value) # plot the quantils of the dataset with the quantils of a gaussian distribution
qqline(exp1$value) # lines the ideal distributaion for a perfect normality of the dataset
```
The plot shows a high amount of expression values equal zero, which has a huge deviation of the ideal gaussian distribution for the expression data. In comparison to the perfect distribution the rest of the data points show a good fit for the regression model. To achieve a better fit the genes that contain expression values equal zero will be removed in the next step.
```{r}
list_all.genes$expression <- list_all.genes$expression[apply(list_all.genes$expression!=0, 1, all),] # removes all rows that contain a value = 0
```
With the melt function the reduced data set is fused to one column of values agian
```{r}
exp1 <- melt.data.frame(list_all.genes$expression, variable_name = "cell line") #fuse all expression values to one long column
```
A q-q-plot is peformed to check the normality of the expression data set once again and determine if the reduction process shows a improved result
```{r}
qqnorm(exp1$value) # plot the quantils of the dataset with the quantils of a gaussian distribution
qqline(exp1$value) # ideal line
```
While the plot indicates a higher accord with the peferct gaussian distribution, there is still a horizontal line pointing out a hogh amount of values near zero. Even if the fit is not perfect, the regression model will be performed with this data points for the gene expression

##### Copy number
As describe in the paragraph above the distribution for the copy number of the genes has to be investigated. First the data is fused to a matrix with two columns
```{r}
cn1 <- melt.data.frame(list_all.genes$copynumber, variable_name = "cellline") # fuse all cell lines
```
Next the distribution is checked with a q-q-plot
```{r}
qqnorm(cn1$value) # create qqplot
qqline(cn1$value) # show ideal line for normality
```
The plot indicates a high deviation for vales approximately smaller than -1, to reduce the influence of this circumstance for the normality, all rows/ genes that show a value smaller than 

### 6.2 Regression model for all driver muatations

### 6.3 Regression model for the single driver mutations

#### 6.3.1 Structuring of the data

We only used the driver mutations, that show a frequency of 5 counts or higher for the linear regression models.
First a new dataframe with expression and copynumber for each driver mutation was established
```{r}
regression.driver <- data.frame(rbind(expression["TP53",], copynumber["TP53",], expression["ARID1A",], copynumber["ARID1A",], expression["ATM",], copynumber["ATM",], expression["BAI1",], copynumber["BAI1",], expression["PTPRF",], copynumber["PTPRF",], expression["SYNE1",], copynumber["SYNE1",], expression["THBS3", ], copynumber["THBS3",] ))
 # new dataframe, which contains expression and copynumber vor TP53 and ARID1Ablubereg
```
For an easy determination between the expression and the copy number of a gene, all the genes receive the index ex or cn. The index ex is used for the expression level while the index cn stands for the copynumber of the gene. 
```{r}
row.names(regression.driver)<- c("TP53ex", "TP53cn", "ARID1Aex", "ARID1Acn", "ATMex", "ATMcn", "BAI1ex", "BAI1cn", "PTPRFex", "PTPRFcn", "SYNE1ex", "SYNE1cn", "THBS3ex", "THBS3cn" ) # rename the rownames, differetiation between ex=expression and cn=copynumber
```
For an easier application the whole dataframe was transposed. Before this process the  columns listed the cell lines and the rows contained the genes.(After the transpose process it is the other way around.)
```{r}
reg.driver <- as.data.frame(t(reg.driver)) # transpose 
```
After a short overview with the view-function, the data frame indicates missing values for the expression level of BAI1. Because this gene can not be included in a regression model, it was removed from the data frame
```{r}
regression.driver <- regression.driver[, -which(colnames(regression.driver) %in% c("BAI1ex", "BAI1cn"))] # remove the columns of the gene BAI1

```

### 6.3.2 Analysis of the correlation and linear regression model

##### TP53
Before using the data of gene TP53 for a linear regression model, a few requirements have to be achieved by the two variables gene expression and copynumber
First both variables should be distributed normally. This circumstances was verified visual with an desity plot and a Q-Q-plot.
```{r}
plot(density(regression.driver$TP53ex), xlab = "expression level", main = "gene TP53") + abline(v = summary(regression.driver$TP53ex)[2:5], col = c("blue", "red", "black", "orange"), lty = 2) # density plot with 1st quantil, median, mean and 3rd quantil
```
Next a qq-plot was performed
```{r}
qqnorm(regression.driver$TP53ex)
qqline(regression.driver$TP53ex)
```
same question?

To check the correlation between the expression and copynumber of the gene TP53 --> scatterplot with correlation
```{r}
plot(TP53cn ~ TP53ex, data = regression.driver, xlab= "expression", ylab = "copynumber", main="TP53", col="blue") + abline(lm(TP53cn~TP53ex, data = regression.driver), col="red") # scatter plot with the line for the pearson correlation coefficient
```
The visual overview of the scatter plot does not conclude a high correlation between the gene expression and the copy number of TP53. To further ceck the correlation, the pearson correlation coefficient was estimated.
```{r}
cor(regression.driver$TP53ex, regression.driver$TP53cn) # compute pearson correlation

```

The value of approximately 0.47 for the pearson correlation coefficient concludes a weak/ moderate positive linear relationship between the two variables. To avoid a strong influence of outliners the spearman correlation to check the correlation a second time. This method will transform the values of each Variable in ranks
```{r}
cor(regression.driver$TP53ex, regression.driver$TP53cn, method = "spearman") # compute spearman correlation

```
The spearman correlation results with approximately 0.37 in an even lower value for the correlation. 

### 6.3 Significance of the correlation coefficient
### 6.4 Creation of a linear regression model
### 6.5 Verfication of the model
## 7. Summary of Results
## 8. Conclusion